pkg/collector/osmlogs_collector.go

```diff
@@ -78,11 +88,19 @@ func (collector *OsmLogsCollector) Collect() error {
 // Helper function to collect output of a given kubectl command to a file
 func collectResourceToFile(collector *OsmLogsCollector, rootPath, fileName string, kubeCmds []string) error {
 	resourceFile := filepath.Join(rootPath, fileName)
-	output, err := utils.RunCommandOnContainer("kubectl", kubeCmds...)
+	outputStreams, err := utils.RunCommandOnContainerWithOutputStreams("kubectl", kubeCmds...)
 	if err != nil {
 		return err
 	}
 
+	output := outputStreams.Stdout
+	// If output is empty, then that means there is no resource of this type within the cluster.
+	// In this case, the stderr output contains the relevant message about not finding the resource (such as "No resource found...").
+	// In this case, write the stderr output to the logfile instead so that the logfile reader knows about the absence of this resource.
```

Comment:
can this be simplified?
```suggestion
	// If stdout output is empty, i.e., there is no resource of this type within the cluster, the absence of this resource is logged in the file with the relevant message from stderr (Ex: "No resource found...").
```